include/pybind11/numpy.h

```diff
@@ -33,7 +34,9 @@
    whole npy_intp / ssize_t / Py_intptr_t business down to just ssize_t for all size
    and dimension types (e.g. shape, strides, indexing), instead of inflicting this
    upon the library user. */
-static_assert(sizeof(ssize_t) == sizeof(Py_intptr_t), "ssize_t != Py_intptr_t");
+static_assert(sizeof(::pybind11::ssize_t) == sizeof(Py_intptr_t), "ssize_t != Py_intptr_t");
+static_assert(std::is_signed<Py_intptr_t>::value, "Py_intptr_t must be signed");
+// We now can reinterpret_cast between py::ssize_t and Py_intptr_t (MSVC + PyPy cares)
```

Comment:
Can we just use `Py_intptr_t` as a scalar type, instead of `py::ssize_t`?

It's uglier, but it should be possible to avoid casting that way.

Reply:
Wouldn't the cast still need to happen somewhere, though? Seems like that would just move the problem?

Reply:
And I think the hope is to do whatever CPython does eventually in PyPy, then remove this.

Reply:
You might be right. I didn't really pay attention how far would we have